worlds/sc2/mission_order/layout_types.py

```diff
@@ -0,0 +1,476 @@
+    """Names of available functions for mission indices. For list member `"my_fn"`, function should be called `idx_my_fn`."""
+
+    def __init__(self, size: int):
+        self.size = size
+
+    def set_options(self, options: Dict[str, Any]) -> Dict[str, Any]:
+        """Get type-specific options from the provided dict. Should return unused values."""
+        return options
+
+    @abstractmethod
+    def make_slots(self, mission_factory: Callable[[], SC2MOGenMission]) -> List[SC2MOGenMission]:
+        """Use the provided `Callable` to create a one-dimensional list of mission slots and set up initial settings and connections.
+
+        This should include at least one entrance and exit."""
+        return []
+    
+    def parse_index(self, term: str) -> Union[Set[int], None]:
+        """From the given term, determine a list of desired target indices. The term is guaranteed to not be "entrances", "exits", or "all".
+
+        If the term cannot be parsed, either raise an exception or return `None`."""
+        return self.parse_index_as_function(term)
+    
+    def parse_index_as_function(self, term: str) -> Union[Set[int], None]:
+        """Helper function to interpret the term as a function call on the layout type, if it is declared in `self.index_functions`.
+        
+        Returns the function's return value if `term` is a valid function call, `None` otherwise."""
+        left = term.find('(')
+        right = term.find(')')
+        if left == -1 and right == -1:
+            # Assume no args are desired
+            fn_name = term.strip()
+            fn_args = []
+        elif left == -1 or right == -1:
+            return None
+        else:
+            fn_name = term[:left].strip()
+            fn_args_str = term[left + 1:right]
+            fn_args = [arg.strip() for arg in fn_args_str.split(',')]
+
+        if fn_name in self.index_functions:
+            try:
+                return getattr(self, "idx_" + fn_name)(*fn_args)
+            except:
+                return None
+        else:
+            return None
+
+    @abstractmethod
+    def get_visual_layout(self) -> List[List[int]]:
+        """Organize the mission slots into a list of columns from left to right and top to bottom.
+        The list should contain indices into the list created by `make_slots`. Intentionally empty spots should contain -1.
+        
+        The resulting 2D list should be rectangular."""
+        pass
+
+class Column(LayoutType):
+    """Linear layout. Default entrance is index 0 at the top, default exit is index `size - 1` at the bottom."""
+
+    # 0
+    # 1
+    # 2
+
+    def make_slots(self, mission_factory: Callable[[], SC2MOGenMission]) -> List[SC2MOGenMission]:
+        missions = [mission_factory() for _ in range(self.size)]
+        missions[0].option_entrance = True
+        missions[-1].option_exit = True
+        for i in range(self.size - 1):
+            missions[i].next.append(missions[i + 1])
+        return missions
```

Comment:
This isn't a bug, they just use indices the same way at the moment, but there's no guarantee this will always be the case. Someone modifying either layout shouldn't have to worry about side effects on the other one.